python/tvm/meta_schedule/builder/local_builder.py

```diff
@@ -0,0 +1,224 @@
+            The signature is Callable[[Module], str].
+        initializer : Optional[Callable[[], None]]
+            The initializer to be used for the worker processes.
+        """
+        super().__init__()
+
+        if max_workers is None:
+            max_workers = cpu_count()
+
+        self.pool = PopenPoolExecutor(
+            max_workers=max_workers,
+            timeout=timeout_sec,
+            initializer=initializer,
+        )
+        self.timeout_sec = timeout_sec
+        self.f_build = f_build
+        self.f_export = f_export
+        self._sanity_check()
+
+    def build(self, build_inputs: List[BuilderInput]) -> List[BuilderResult]:
+        results: List[BuilderResult] = []
+        map_result: MapResult
+
+        # Dispatch the build inputs to the worker processes.
+        for map_result in self.pool.map_with_error_catching(
+            lambda x: LocalBuilder._worker_func(*x),
+            [
+                (
+                    self.f_build,
+                    self.f_export,
+                    build_input.mod,
+                    build_input.target,
+                )
+                for build_input in build_inputs
+            ],
+        ):
+            if map_result.status == StatusKind.COMPLETE:
+                results.append(BuilderResult(map_result.value, None))
+            elif map_result.status == StatusKind.TIMEOUT:
+                results.append(
+                    BuilderResult(
+                        None,
+                        f"LocalBuilder: Timeout, killed after {self.timeout_sec} seconds",
+                    )
+                )
+            elif map_result.status == StatusKind.EXCEPTION:
+                results.append(
+                    BuilderResult(
+                        None,
+                        "LocalBuilder: An exception occurred\n" + str(map_result.value),
+                    )
+                )
+            else:
+                raise NotImplementedError(map_result.status)
+        return results
+
+    def _sanity_check(self) -> None:
+        def _check(
+            f_build: Optional[str],
+            f_export: Optional[str],
+        ) -> None:
+            get_global_func_with_default_on_worker(name=f_build, default=None)
+            get_global_func_with_default_on_worker(name=f_export, default=None)
+
+        value = self.pool.submit(_check, self.f_build, self.f_export)
+        value.result()
+
+    @staticmethod
+    def _worker_func(
+        _f_build: Optional[str],
+        _f_export: Optional[str],
+        mod: IRModule,
+        target: Target,
+    ) -> str:
+        # Step 0. Get the registered functions
+        f_build: Callable[
+            [IRModule, Target],
+            Module,
+        ] = get_global_func_with_default_on_worker(_f_build, default_build)
+        f_export: Callable[
+            [Module],
+            str,
+        ] = get_global_func_with_default_on_worker(_f_export, default_export)
```

Comment:
nit: You could use custom type such as `FBuildType = Callable[[IRModule, Target], Module,]` to avoid redundant complex type strings.

Reply:
Yeah it is good point too. In the latest commit, I added LocalBuilder.T_BUILD and its corresponding documentation